src/redis-cli.c

```diff
@@ -9184,18 +9290,43 @@ static void findBigKeys(int memkeys, unsigned memkeys_samples) {
                     exit(1);
                 }
 
-                printf(
-                   "[%05.2f%%] Biggest %-6s found so far '%s' with %llu %s\n",
-                   pct, type->name, type->biggest_key, sizes[i],
-                   !memkeys? type->sizeunit: "bytes");
+                /* We only show the original progress output when writing to a file */
+                if (!(isatty(STDOUT_FILENO) || getenv("FAKETTY"))) {
+                    printf("[%05.2f%%] Biggest %-6s found so far %s with %llu %s\n",
+                        pct, type->name, type->biggest_key, sizes[i],
+                        !memkeys? type->sizeunit: "bytes");
+
+                    /* Update overall progress */
+                    if(sampled % 1000000 == 0) {
+                        printf("[%05.2f%%] Sampled %llu keys so far\n", pct, sampled);
+                    }
+                }
 
                 /* Keep track of the biggest size for this type */
                 type->biggest = sizes[i];
             }
 
-            /* Update overall progress */
-            if(sampled % 1000000 == 0) {
-                printf("[%05.2f%%] Sampled %llu keys so far\n", pct, sampled);
+            /* Show the progress bar in TTY */
+            if (mstime() > refresh_time + REFRESH_INTERVAL && 
+                (isatty(STDOUT_FILENO) || getenv("FAKETTY"))) {
+                int line_count = 0;
```

Comment:
```suggestion
            if (mstime() > refresh_time + REFRESH_INTERVAL &&
                (isatty(STDOUT_FILENO) || getenv("FAKETTY")))
            {
                int line_count = 0;
```

Reply:
Thanks, I will correct the same mistake I did in updateTopSizes(), keyStats(), and findHotKeys().